### ISSUE TITLE
Dropdown: Field 'Required' is not read by screen reader

### ISSUE THREAD

**Issue by v-neetsi** [OfficeDev/office-ui-fabric-react#5162]
Bug Report:
Package version(s): (5.66.0)
Browser and OS versions: Edge, Screen Reader(Narrator)
Priorities and help requested (not applicable if asking question):
Are you willing to submit a PR to fix? (No)
Requested priority: ( Normal)
Products/sites affected: (if applicable)

Describe the issue:
For implementation of 'Dropdown' using 'required' prop with 'label' prop, dropdown label is showing the (*) but narrator does not read Dropdown as 'Required' when dropdown receives focus. Similar case with Date Picker.

The behavior should be similar like TextField.

Actual behavior:
Screen reader is not reading dropdown as required (if 'required' prop is specified) when it receives focus.

Expected behavior:
If 'required' prop is set for dropdown then Screen Reader should read Dropdown as 'Required' when focus reaches to this(Similar to TextField)
If applicable, please provide a codepen repro:

**Comment by chshrikh** [comment]
we are observing same issue with "inspect". Inspect is not able to detect the drop down as required and marks "IsRequiredForForm" as false

**Comment by micahgodbolt** [comment]
@v-neetsi from looking at the markup, any idea why narrator isn't picking up the required value? The label has the proper 'for' connected to the dropdown, and the dropdown div has the 'required' value on it. 


https://codepen.io/micahgodbolt/pen/GBoXxb

Is it an issue of required not working properly on a div? Thoughts on how this can be resolved? Happy to make the code change if you know what change should be made.

**Comment by arunarokk** [comment]
@micahgodbolt , I searched many examples , aria-required only working with select or input elements, I have verified your codepen snippet with jaws on chrome, its not reading required for dropdown.

**Comment by micahgodbolt** [comment]
it's possible that we don't have a solution for this problem. If a select/input element is required, we'll have a hard time making this work. The move to div was necessary to support the customized functionality. Is this a problem then with jaws and other screenreaders?

**Comment by arunarokk** [comment]
I have tested dropdown with narrator & jaws, where its not reading required. for this "it's possible that we don't have a solution for this problem", i need to discuss this with team and let you know.

**Comment by arunarokk** [comment]
As an alternative, using label to announce if the dropdown is a required. you can close the issue. thanks

**Comment by micahgodbolt** [comment]
can you demo how that works? I want to make sure we capture that as best practices.

**Comment by cliffkoh** [comment]
> As an alternative, using label to announce if the dropdown is a required. you can close the issue. thanks

Closing based on conclusion reached in discussion above.

**Comment by satbai** [comment]
@cliffkoh @arunarokk  I can still reproduce this.

For a workaround, adding both required="true" and aria-required="true" for the Dropdown component props will fix this but I think this should be fixed in the component so that if required="true" is provided, aria-required value will be automatically added as well. Could this be reconsidered please?

**Comment by micahgodbolt** [comment]
@satbai can you create a new issue for that explicit ask? If it's possible to do without breaking anything, we can certainly get it into our backlog.

**Comment by gabrielaalfaro** [comment]
I used both required, and aria-required, but fot he Dropdown component the narrator do not read REQUIRED
I have to remove the label property, only in that case it works, but that property is a must to have to me.

**Comment by cliffkoh** [comment]
Hi @gabrielaalfaro, the issue was closed based on what @arunarokk said above. I'll reopen so this issue can be triaged with a conclusion properly.

**Comment by dzearing** [comment]
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute

Perhaps we need to add aria-required?

**Comment by gabrielaalfaro** [comment]
i used the aria required attribute but is not fixing the problem

**Comment by natalieethell** [comment]
I'm wondering if aria-required is only read out if it's placed on an input element. I'm having no trouble hearing Narrator on Edge read out "required" when adding it to an input element, but it's not read out on a div element (yet other aria attributes are read out by Narrator for the div element). Here's a CodePen demonstrating this behavior: https://codepen.io/naethell/pen/pQzzZw.

Adding a required label might be the best approach here.

**Comment by natalieethell** [comment]
Similarly, the `required` attribute works well on input elements (in the TextField case), but not on div elements.

**Comment by cliffkoh** [comment]
I took a look at @natalieethell's codepen and verified that Edge does not do anything with the `aria-required` attribute on <div>. As such, Narrator would not read it as being required.

The spec for `aria-required` does not indicate what elements should support it - all examples are of `<input>`, as far as I can tell.
https://www.w3.org/TR/WCAG20-TECHS/ARIA2.html

**Comment by natalieethell** [comment]
Just waiting on some feedback from the accessibility PM. 

Here are some updates:

When adding `aria-hidden` to the top-most element in addition to `aria-required`, Narrator does read out “required”, however that does leave out the other aria attributes, such as `aria-describedby` and `aria-activedescendant`, which we need the screen reader to read out for dropdown.

Here’s a CodePen with two examples:

https://codepen.io/naethell/pen/WYdopo 

The first has `aria-required` set to true on the focusable element, and “required” is not read out. Note also that the “Required Dropdown” is just the label text.

**Required Dropdown, Option d, <say-as interpret-as="spell">4</say-as> of <say-as interpret-as="spell">7</say-as>, selected,** 

The second has `aria-required` and `aria-hidden` set to true on the focusable element, and “required” is read out, but the `aria-describedby`/`aria-activedescendant` (“Option d”) is not. It also says there are 0 items in the dropdown.

**Required Dropdown with `aria-hidden`, selected, requires selection contains 0 items,Required, 
collapsed,**

**Comment by natalieethell** [comment]
Update: I have a PR out that should hopefully fix the issue!
